### PR TITLE
Add triangle to convexity helper

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
# PR Description

Add specific methods in ConvexityHelper and DigitalConvexity to build lattice polytope from 2 or 3 points, even in degenerate cases. This speeds up their construction by a factor 3 to 5.

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug mode.
- [x] All continuous integration tests pass (Github Actions)

### CODE REVIEW

**Comment by JacquesOlivierLachaud** [comment]
PythonTests fail but it seems unrelated to this PR.

File: ChangeLog.md [hunk 1]
```diff
@@ -73,14 +73,17 @@
...
 
-
 - *Topology package*
   - Fix KhalimskySpaceND to get it work with BigInteger (Tristan Roussillon,
     [#1681](https://github.com/DGtal-team/DGtal/pull/1681)
 
 - *Geometry package*
   - Fix Issue #1676 in testStabbingCircleComputer (Tristan Roussillon,
     [#1688](https://github.com/DGtal-team/DGtal/pull/1688)
+  - Add creation of polytopes from segment and triangles in
```
Comment (dcoeurjo) [inline ChangeLog.md:83]:
```suggestion
  - Add creation of polytopes from segments and triangles in
```

**Review by dcoeurjo — APPROVED** [review]
Looks good to me, just a minor comment

**Comment by dcoeurjo** [comment]
It looks like "testConvexityHelper" fails in the CI bots.

**Comment by JacquesOlivierLachaud** [comment]
Curiously it is working on my Mac Debug/Release. Do you have an ubuntu to tell me what's happening ?

**Comment by dcoeurjo** [comment]
On a linux docker image:
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
testConvexityHelper is a Catch v2.13.7 host application.
Run with -? for options

-------------------------------------------------------------------------------
Scenario: ConvexityHelper< 3 > triangle tests
      Given: Given non degenerated triplets of points
       When: Computing their tightiest polytope or triangle, dilated by a cube
       Then: They have the same number of interior points
-------------------------------------------------------------------------------
/DGtal/tests/geometry/volumes/testConvexityHelper.cpp:528
...............................................................................

/DGtal/tests/geometry/volumes/testConvexityHelper.cpp:529: FAILED:
  REQUIRE( nbi_ok == nb_total )
with expansion:
  18 == 19
with messages:
  a := (4, 3, 9)
  b := (6, 4, 6)
  c := (7, 4, 6)
  n := (0, -3, -1)
  P := [BoundedLatticePolytope<3> A.rows=12 valid_edge_constraints=1]
    [ 1 0 0 ] . x <= 8
    [ -1 0 0 ] . x <= -4
    [ 0 1 0 ] . x <= 5
    [ 0 -1 0 ] . x <= -3
    [ 0 0 1 ] . x <= 10
    [ 0 0 -1 ] . x <= -6
    [ 0 -3 -1 ] . x <= -18
    [ 0 3 1 ] . x <= 22
    [ -3 0 -2 ] . x <= -30
    [ -1 2 0 ] . x <= 4
    [ 3 0 3 ] . x <= 45
    [ 1 -3 0 ] . x <= -4
  T := [BoundedLatticePolytope<3> A.rows=14 valid_edge_constraints=1]
    [ 1 0 0 ] . x <= 8
    [ -1 0 0 ] . x <= -4
    [ 0 1 0 ] . x <= 5
    [ 0 -1 0 ] . x <= -3
    [ 0 0 1 ] . x <= 10
    [ 0 0 -1 ] . x <= -6
    [ 0 -3 -1 ] . x <= -18
    [ 0 3 1 ] . x <= 22
    [ -3 0 -2 ] . x <= -30
    [ 0 0 -1 ] . x <= -6
    [ 3 0 3 ] . x <= 45
    [ -1 2 0 ] . x <= 4
    [ 0 1 0 ] . x <= 5
    [ 1 -3 0 ] . x <= -4
  nb_P := 24
  nb_T := 24
  nbi_P := 8
  nbi_T := 6

===============================================================================
test cases:   4 |   3 passed | 1 failed
assertions: 103 | 102 passed | 1 failed
```

**Comment by dcoeurjo** [comment]
Hi @JacquesOlivierLachaud , were you able to reproduce the bug ?

**Comment by JacquesOlivierLachaud** [comment]
Sorry I haven't found any time to work on it (and I forgot at some point :) ) . I will come back to it soon.

**Comment by dcoeurjo** [comment]
I'm going to create a DGtal release, would you like this PR to be included?

**Comment by JacquesOlivierLachaud** [comment]
What is the deadline ? I might be able to do it next week.

**Comment by dcoeurjo** [comment]
I'd say by Wednesday ?

**Comment by JacquesOlivierLachaud** [comment]
When I build it on linux noble docker image in Debug Mode, it passes ! Only testLinearStructure fails:
```
Output:
----------------------------------------------------------
^[[0m^[[1mNew Block [testing 3d operators]^[[0m
  [dec | primal 0-cells <-> dual 3-cells (2) | primal 1-cells <-> dual 2-cells (5) | primal 2-cells <-> dual 1-cells (4) | primal 3-cells <-> dual 0-cells (1)]
  ^[[0m^[[1mNew Block [base operators]^[[0m
    d0
[primal 0-form => primal 1-form (2 => 5)]
 1  0
 0 -1
 0 -1
 1 -1
 1  0
    d2p
[dual 2-form => dual 3-form (5 => 2)]
 1  0  0  1  1
 0 -1 -1 -1  0
    ^[[0m^[[31m[ERR] Fatal Error - assertion (Eigen::MatrixXd(d0.myContainer) == d0_th) failed in void test_manual_operators_3d(): /tmp/DGtal/tests/dec/testLinearStructure.cpp(391)
<end of output>
Test time =   0.00 sec
```

**Comment by dcoeurjo** [comment]
👌 let me try again (I was able to reproduce the bug)

**Comment by JacquesOlivierLachaud** [comment]
Oups, I have made a mistake. I think I am not on the correct branch.

**Comment by JacquesOlivierLachaud** [comment]
Sorry. I can reproduce it. I try to look at it today (but tonight probably).

**Comment by dcoeurjo** [comment]
ok great. Here you have my trace: 
    https://asciinema.org/a/RYYW8Mm3iupP70Jo01zuP6lmK

**Comment by dcoeurjo** [comment]
(ubuntu jammy)

**Comment by JacquesOlivierLachaud** [comment]
It looks like the other errors are unrelated to this PR.

**Review by dcoeurjo — APPROVED** [review]
